### PR TITLE
nix: switch to modules to populate Hercules CI's onSchedule jobs

### PR DESCRIPTION
**Pull request by ConnorBaker**
I started thinking about what we need to know to run the jobs as they are currently. Essentially only a handful of things:

- frequency: handled by the `when` attribute set
- branch: what to build against
- system: `x86_64-linux` or `aarch64-linux` for Jetson
- CUDA capabilities and forward compatibility: what device code to generate
- `jobsAttr`: whether we're building checks or packages which should never be broken
- reason: some sort of text describing _why_ we're doing the build

In an earlier version of this refactor, I used lists of attribute sets instead of named attribute sets. This earlier version then generated names for each attribute set in the list using the values it supplied. I decided against that for the final version because I think it's important to be able to easily trace back the name of the job to the code (that is, the job name shouldn't be generated at eval time).

I've not tested this on Hercules (nor do I know how to) but poking around in the REPL seems to work fine.

### CODE REVIEW

File: nix/ci/default.nix [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+            type = types.submodule {
+              options = {
+                dayOfWeek = mkOption {
+                  description = "Days of week to build";
+                  type = types.nonEmptyListOf types.str;
+                };
+                hour = mkOption {
+                  description = "Hours of day to build";
+                  type = types.nonEmptyListOf types.ints.positive;
+                };
```
Comment (SomeoneSerge) [inline nix/ci/default.nix]: I guess this is more like `oneOf`? Or we follow upstream and use `default = null`: https://github.com/hercules-ci/hercules-ci-effects/blob/31b6cd7569191bfcd0a548575b0e2ef953ed7d09/flake-modules/types/when.nix#L35-L52

The related eval error: https://hercules-ci.com/github/SomeoneSerge/nixpkgs-cuda-ci/jobs/7346

**Comment by SomeoneSerge** [comment]
Thanks!

> jobsAttr: whether we're building checks or packages which should never be broken

I've been really looking forward to remove it, and now that it's been better formalized that should get much easier to do

File: nix/ci/default.nix (hunk 1 above)
Comment (ConnorBaker) [reply nix/ci/default.nix]: I don't know why I didn't just re-use upstream's definition in the first place -- fixed in latest.

**Comment by SomeoneSerge** [comment]
I rebased after some changes in the `flake.nix` (frequencies mostly), and renamed a few attributes. Merging in the current form as we can always follow up. Thank you so much for pushing this!

**Pull request merged**